src/Http/RequestFactory.php

```diff
@@ -218,4 +231,38 @@ public function createHttpRequest()
 		return new Request($url, NULL, $post, $files, $cookies, $headers, $method, $remoteAddr, $remoteHost, $rawBodyCallback);
 	}
 
+	/**
+	 * @param bool $usingTrustedProxy
+	 *
+	 * @return string 'http' or 'https'
+	 */
+	private function getScheme($usingTrustedProxy)
+	{
+		if ($usingTrustedProxy) {
+			if (!empty($_SERVER['HTTP_X_FORWARDED_PROTO'])) {
+				if (strcasecmp($_SERVER['HTTP_X_FORWARDED_PROTO'], 'https') === 0) {
+					return 'https';
+				}
+
+				return 'http';
+			}
+
+			if (!empty($_SERVER['HTTP_X_FORWARDED_PORT'])) {
+				$port = (int) $_SERVER['HTTP_X_FORWARDED_PORT'];
+
+				if ($port === 443) {
+					return 'https';
+				} elseif ($port === 80) {
+					return 'http';
+				}
+			}
```

Comment:
AFAIK you cannot derive schema from port. How does Symfony and Zend handle this?